resources/knative-eventing-kafka/templates/uninstall.yaml

```diff
@@ -0,0 +1,107 @@
+# Copyright 2019 The Kyma Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# About:
+# This pre-delete job removes all kafkachannel custom resources
+# It does so by using a job which is executed before the helm chart is deleted
+
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: {{ .Release.Name }}-cr-delete
+  namespace: {{ .Release.Namespace }}
+  annotations:
+    # Executes on a deletion request before any resources are deleted from Kubernetes
+    # this gives the controllers a chance to act on CRs with a finalizer
+    # source: https://helm.sh/docs/topics/charts_hooks/
+    helm.sh/hook: pre-delete
+    helm.sh/hook-delete-policy: "hook-succeeded,before-hook-creation"
+    # everything without a dependency has weight 0
+    helm.sh/hook-weight: "0"
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: {{ .Release.Name }}-cr-delete
+  annotations:
+    helm.sh/hook: pre-delete
+    helm.sh/hook-delete-policy: "hook-succeeded,before-hook-creation"
+    helm.sh/hook-weight: "0"
+rules:
+- apiGroups: ["knativekafka.kyma-project.io"]
+  resources: ["kafkachannels"]
+  verbs: ["list", "delete"]
+- apiGroups: ["apiextensions.k8s.io"]
+  resources: ["customresourcedefinitions"]
+  verbs: ["list"]
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: {{ .Release.Name }}-cr-delete
+  annotations:
+    helm.sh/hook: pre-delete
+    helm.sh/hook-delete-policy: "hook-succeeded,before-hook-creation"
+    helm.sh/hook-weight: "0"
```

Comment:
Since it depends on ServiceAccount and Clusterrole hook-weight of 1 is more reliable.